controllers/reconcile.go

```diff
@@ -475,6 +577,103 @@ func populateVolumes(d *apps.Deployment, frontend *crd.Frontend, frontendEnviron
 	d.Spec.Template.Spec.Volumes = volumes
 }
 
+// ObjectStoreBucket represents the configuration for an object storage bucket.
+type ObjectStoreBucket struct {
+	AccessKey *string
+	SecretKey *string
+	Name      string
+	Region    *string
+	Endpoint  *string
+	Port      *string
+	TLS       *bool
+}
+
+// ExtractBucketConfigFromSecret extracts ObjectStoreBucket configuration for a specific bucket name
+// from a given Kubernetes Secret. It searches for the bucket name either directly
+// in the secret's Data map under the "bucket" key, or within the
+// "clowder/bucket-names" annotation (if it's a comma-separated list).
+//
+// It returns the ObjectStoreBucket configuration and an error if the secret data is
+// invalid or incomplete for the specified bucket, or if the bucket name isn't found.
+func ExtractBucketConfigFromSecret(secrets []v1.Secret, targetBucketName string) (*ObjectStoreBucket, error) {
+	if len(secrets) == 0 {
+		return nil, fmt.Errorf("no secrets provided to search for bucket '%s'", targetBucketName)
+	}
+
+	clowderBucketNamesAnnotation := "clowder/bucket-names"
```

Comment:
This is interesting. How does Clowder fit here? AFAIK, it's its own operator. Is this just a silly naming that is only confusing and we have to deal with it?

Reply:
Yep, unfortunately, it looks like app-sre uses this annotation to store some bucket names: https://github.com/RedHatInsights/clowder/blob/82ae0543e41df4c26acd4aee6f3a3679c261c4f6/controllers/cloud.redhat.com/providers/objectstore/appinterface.go#L119